Document ``catch Panic(uint)``. (#10682)
ethereum/solidity: docs/control-structures.rst

::

    // SPDX-License-Identifier: GPL-3.0

>     pragma solidity ^0.8.1;

Reviewer: Maybe will fix failure:
```suggestion
    pragma solidity >0.8.0;
```

Author: I think we should fix the script: If the release is not available, use the current branch or something like that.

Reviewer 2: I see you added a workaround specifically to  be able to use non-existent `.99` versions: b03d1105289eb3fb53dfbf58916aed56a13cf3e6 :)

So either use that or we could just remove the `exit 1` from the script and it would just skip these errors.

If you want, we could also make it ignore version one higher than the latest release - it does download `list.txt` from `solc-bin` so we have all the necessary information to determine that.

Author: We cannot use `.99` here. I want to use the next non-breaking release that is not released yet.

Author: We could just add a special case that assumes that the version of the current branch / the develop branch has been released.

Reviewer 2: Oh, right. I didn't look at the example closely.

Well, sure, a special case for version from develop sounds pretty easy. I can do that. I'll submit a PR tomorrow.

Reviewer 2: My PR that solves the problem:  #10770. You can now rebase your PR on it and see if it works.

these variables are assigned and the contract's execution continues inside the
first success block. If the end of the success block is reached, execution continues after the ``catch`` blocks.

Solidity supports different kinds of catch blocks depending on the
type of error:

- If the error was caused by ``revert("reasonString")`` or
  ``require(false, "reasonString")`` (or an internal error that causes such an
  exception), then the catch clause
  of the type ``catch Error(string memory reason)`` will be executed.

- If the error was caused by a panic, i.e. by a failing ``assert``, division by zero,
  invalid array access and others, the catch clause of the type
  ``catch Panic(uint errorCode)`` will be run.

> - The clause ``catch (bytes memory lowLevelData)`` is executed if the error signature

Reviewer 3: ```suggestion
The clause ``catch (bytes memory lowLevelData)`` is executed if the error signature
```

Author: This is the third kind, so I think it's fine to use the bullet point. I'll check if I can rephrase it so it's more obvious.

  does not match any other clause, if there was an error while decoding the error
  message, or
  if no error data was provided with the exception.

>   The declared variable provides access to the low-level error data in that case.

Reviewer 3: Not sure if the single whitespace here and the lines before is intended?

  If you are not interested in the error data, you can just use

>   ``catch { ... }`` (even as the only catch clause).

Reviewer 3: Is the indentation here and in the previous line intended?

It is planned to support other types of error data in the future.

> The strings ``Error`` and ``Panic`` are currently parsed as is and is not treated as an identifier.

Reviewer 3: ```suggestion
The strings ``Error`` and ``Panic`` are currently parsed as is and are not treated as identifiers.
```
